src/WebJobs.Script/ScriptHostBuilderExtensions.cs

```diff
@@ -411,6 +411,26 @@ internal static void ConfigureApplicationInsights(HostBuilderContext context, IL
                         o.EnableDependencyTracking = false;
                     });
                 }
+
+                builder.Services.AddOptions<LoggerFilterOptions>().Configure<IEnvironment>((options, environment) =>
+                {
+                    // Skip sending user generated logs to AI and QuickPulse if worker AI agent is configured, worker will send these logs to AI and Quickpulse service.
+                    if (environment.IsApplicationInsightsAgentEnabled())
+                    {
+                        options.AddFilter<ApplicationInsightsLoggerProvider>((category, logLevel) =>
```

Comment:
Nothing against the filter, but curious what the reasoning for filtering here vs in a `ITelemetryProcessor` - especially since this is AppInsights specific.

Reply:
ITelemetryProcessor processes the telemetry information before it is sent to Application Insights whereas AddFilter will filter this earlier in the pipeline. This should save some compute.